pkg/ttnpb/udp/packet_data.go

```diff
@@ -128,6 +147,11 @@ type Stat struct {
 	FPGA *uint32       `json:"fpga,omitempty"` // Version of Gateway FPGA (unsigned integer)
 	DSP  *uint32       `json:"dsp,omitempty"`  // Version of Gateway DSP software (unsigned interger)
 	HAL  *string       `json:"hal,omitempty"`  // Version of Gateway driver (format X.X.X)
+	HVer *struct {
+		FPGA *uint32 `json:"fpga,omitempty"` // Version of FPGA (unsigned integer)
+		DSP0 *uint32 `json:"dsp0,omitempty"` // Version of DSP 0 software (unsigned interger)
+		DSP1 *uint32 `json:"dsp1,omitempty"` // Version of DSP 1 software (unsigned interger)
+	} `json:"hver,omitempty"` // Gateway hardware versions
```

Comment:
```suggestion
	DSP  *uint32       `json:"dsp,omitempty"`  // Version of Gateway DSP software (unsigned integer)
	HAL  *string       `json:"hal,omitempty"`  // Version of Gateway driver (format X.X.X)
	HVer *struct {
		FPGA *uint32 `json:"fpga,omitempty"` // Version of FPGA (unsigned integer)
		DSP0 *uint32 `json:"dsp0,omitempty"` // Version of DSP 0 software (unsigned integer)
		DSP1 *uint32 `json:"dsp1,omitempty"` // Version of DSP 1 software (unsigned integer)
	} `json:"hver,omitempty"` // Gateway hardware versions
```

Where does `hver` come from, where is it specified?

I seem to remember there were fields very specific to Kerlink gateways, which we may not want to include here. If these are (still proprietary) generic UDP packet forwarder fields, then I'm okay adding them.

It's just that any mistakes here, because some (future) packet forwarders use a different data type or anything, will result in unmarshaling errors. They're not worth it because nobody really uses this data I believe.

Reply:
The Kerlink Common Packet Forwarder will indeed fill the top level `dsp` and `hal` fields. You can find their definition of the fields [here](https://github.com/TheThingsIndustries/lorawan-stack/issues/3400#issuecomment-1227698944). It is indeed the case that the public source code of `packet_forwarder` [does not include them](https://github.com/Lora-net/packet_forwarder/blob/d0226eae6e7b6bbaec6117d0d2372bf17819c438/PROTOCOL.TXT#L214-L244).

The `hver` field is indeed part of the still proprietary [packet forwarder statistics message definition](https://github.com/TheThingsIndustries/lorawan-stack/issues/3400#issuecomment-1227700752).

Reply:
Okay, well if it's worth unmarshaling errors because of other gateways that use the same fields but with different types, I'm fine with it. Let's do add metrics though for UDP unmarshaling errors so at least we can see how much traffic we drop.

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/5727/commits/0257604af8920b0d0479f3709432a30e308a8cea should do it. It will also record the field which has a type mismatch and the observed value in a minimal fashion - see [the docs](https://pkg.go.dev/encoding/json#UnmarshalTypeError).